Automattic/pocket-casts-android
# Developer settings clean up

## Description

This pull request makes the following changes:
- Rewrites the Developer page in Compose. 
- Adds a preview to Showkase for the RowButton disabled and without padding.
- Fixes the Showkase preview for the UserPasswordFields in the dark theme.
- Adds the Text components to Showkase.
- Changes the SettingRow component to accept a Painter rather than a GradientIconData. It felt a bit strange always passing a GradientIconData and it means we can use Compose ImageVectors at a later date.

As part of the Developer page rewrite I removed the actions "Sync up next" and "Launch billing dialog" as they were no longer being used.

## Testing Instructions
1. Tap the Profile tab
2. Tap the settings cog
3. Tap the Developer section
4. ✅ Verify the Developer looks correct

## Screenshots

| Before | After |
| --- | --- |
| ![Screenshot_20230712_115845](https://github.com/Automattic/pocket-casts-android/assets/308331/574a747a-619c-47d4-930e-53b816d9e0f6) | ![Screenshot_20230712_161700](https://github.com/Automattic/pocket-casts-android/assets/308331/e5dbce80-05bb-4878-bc21-ab68ea04608b) |
